packages/qwik/src/core/use/use-resource.ts

```diff
@@ -204,10 +204,15 @@ function getResourceValueAsPromise<T>(props: ResourceProps<T>): Promise<JSXOutpu
         }
       }
     }
-    return resource.value.then(
-      useBindInvokeContext(props.onResolved),
-      useBindInvokeContext(props.onRejected)
-    );
+    const value = resource.value;
+    if (value) {
+      return value.then(
+        useBindInvokeContext(props.onResolved),
+        useBindInvokeContext(props.onRejected)
+      );
+    } else {
+      return Promise.resolve(undefined);
```

Comment:
can you put a comment here explaining what happens next? It seems as if that is the end of the resource loading?